cmd/embedded-cluster/install.go

```diff
@@ -551,31 +554,42 @@ func runOutro(c *cli.Context, applier *addons.Applier, cfg *k0sconfig.ClusterCon
 func maybeAskAdminConsolePassword(c *cli.Context) (string, error) {
 	defaultPassword := "password"
 	userProvidedPassword := c.String("admin-console-password")
-	if c.Bool("no-prompt") {
-		if userProvidedPassword != "" {
-			return userProvidedPassword, nil
-		} else {
-			logrus.Infof("The Admin Console password is set to %s", defaultPassword)
-			return defaultPassword, nil
-		}
-	}
+	// If there's a user provided password we'll try that first
 	if userProvidedPassword != "" {
+		if !validateAdminConsolePassword(userProvidedPassword, "", false) {
+			return "", fmt.Errorf("unable to set the Admin Console password")
+		}
```

Comment:
I'm unsure if we only wanna validate the prompts on the case where the user provides a password through a CLI param. I assumed the latter.